### PR TITLE
Added capturing of MDC context, fixed small typos

### PR DESCRIPTION
**Pull request by AkaZver**
Hello!

I was trying to use your cool library in my tests and didn't find one important feature - capturing of MDC context.
So I decided to add it myself :)
Give me some feedback please.

Thank you

### CODE REVIEW

File: src/main/java/nl/altindag/log/LogCaptor.java [hunk 1]
```diff
@@ -29,12 +29,7 @@
 import java.time.Instant;
 import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (Hakky54) [inline src/main/java/nl/altindag/log/LogCaptor.java:32]: I prefer to explicitly list all the used objects instead of the wildcard. It just just a personal preference... Would it be possible for you to revert this change?

File: src/main/java/nl/altindag/log/LogCaptor.java [hunk 2]
```diff
@@ -147,6 +142,7 @@ private static LogEvent toLogEvent(ILoggingEvent iLoggingEvent) {
         String level = iLoggingEvent.getLevel().toString();
         String loggerName = iLoggingEvent.getLoggerName();
         ZonedDateTime timeStamp = ZonedDateTime.ofInstant(Instant.ofEpochMilli(iLoggingEvent.getTimeStamp()), ZoneOffset.UTC);
+        Map<String, String> mdcProperties = iLoggingEvent.getMDCPropertyMap();
```
Comment (Hakky54) [inline src/main/java/nl/altindag/log/LogCaptor.java:145]: A small detail maybe... I prefer not to use abbreviations as it would be more difficult the understand. The getMDCPropertyMap just returns a `map` containing the `mapped diagnostic context`. What do you think of the variable name `diagnosticContext`? It is just a shorter version of `MDC` without the `Mapper`

Comment (Hakky54) [reply src/main/java/nl/altindag/log/LogCaptor.java:145]: The method `iLoggingEvent.getMDCPropertyMap()` returns a mutable map. It is better to wrap it into a immutable map by using `Collections.unmodifiableMap()`

File: src/main/java/nl/altindag/log/LogCaptor.java (hunk 1 above)
Comment (AkaZver) [reply src/main/java/nl/altindag/log/LogCaptor.java]: Done

**Comment by Hakky54** [comment]
Hey @AkaZver 

Thank you very much for this PR. I like your idea of adding this feature, I was not aware of it because I never used it.
I am just adding couple of comments. Most of it are just minor details. Can you maybe take a look at it?

File: README.MD [hunk 3]
```diff
@@ -284,7 +284,7 @@ public class FooServiceShould {
 ```
 
 ## Using Log Captor alongside with other logging libraries
-When building your maven or gradle project it can complain that you are using multiple SLF4J implementations. Log Captor is using logback as SLF4J implementation and SLF4J doesn't allow you to use multiple implementations, therefor you need to explicitly specify which to use during which build phase.
+When building your maven or gradle project it can complain that you are using multiple SLF4J implementations. Log Captor is using logback as SLF4J implementation and SLF4J doesn't allow you to use multiple implementations, therefore you need to explicitly specify which to use during which build phase.
```
Comment (Hakky54) [inline README.MD:287]: Nice catch 👍🏼

File: src/main/java/nl/altindag/log/LogCaptor.java (hunk 2 above)
Comment (AkaZver) [reply src/main/java/nl/altindag/log/LogCaptor.java]: Looks ok :)
Done x2

Comment (Hakky54) [reply src/main/java/nl/altindag/log/LogCaptor.java]: Thank you very much!

File: src/test/java/nl/altindag/log/LogCaptorShould.java [hunk 4]
```diff
@@ -359,6 +361,37 @@ void filterInfoMessages() {
                 .allMatch(Level.INFO::equals, "INFO");
     }
 
+    @Test
+    void captureMdcHeadersWhereLog4jIsUsed() {
+        logCaptor = LogCaptor.forClass(ServiceWithApacheLog4jAndMdcHeaders.class);
+
+        new ServiceWithApacheLog4jAndMdcHeaders().sayHello();
```
Comment (Hakky54) [inline src/test/java/nl/altindag/log/LogCaptorShould.java:368]: This is a small detail, and again a personal preference... I like the inline way of calling it but would it be possible to have it similar to the other unit tests? So something like:
```
Service service = new ServiceWithApacheLog4jAndMdcHeaders();
service.sayHello();
```

Comment (AkaZver) [reply src/test/java/nl/altindag/log/LogCaptorShould.java]: Done

**Comment by Hakky54** [comment]
Thank you very much Vasiliy, this is a nice feature. I will publish it right away so you can use it straight away

**Comment by AkaZver** [comment]
> Thank you very much Vasiliy, this is a nice feature. It looks good to me right now. I will publish it right away so you can use it straight away

Thank you for your work :)

**Comment by Hakky54** [comment]
Today version 2.6.0 is released with the Diagnostic Context
